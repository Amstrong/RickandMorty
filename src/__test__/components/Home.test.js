import React from "react";
import { mount } from "enzyme";
import { Link } from "react-router-dom";
import renderer from "react-test-renderer";
import { StaticRouter } from "react-router";
import Home from "../../frontend/pages/home/homePage";
import ProviderMock from "../../__mocks__/ProviderMock";

describe("Test del Home", () => {
  test("Home: Comprobando la UI", () => {
    const home = mount(
      <ProviderMock>
        <Home />
      </ProviderMock>
    );
    expect(home.find(".text--title").text()).toEqual("Rick&Morty");
  });

  test("Home: Comprobando Link de ReactRouter", () => {
    const component = renderer.create(
      <StaticRouter location="someLocation" context={{}}>
        <Link to="/characters" />
      </StaticRouter>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
