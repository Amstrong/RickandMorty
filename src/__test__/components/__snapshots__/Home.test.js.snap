// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Snapshot Comprobar la UI del component Home 1`] = `
<div
  className="container"
>
  <div
    className="photo"
  />
  <div
    className="container__elements"
  >
    <p
      className="text--title"
    >
      Rick&Morty
    </p>
    <p
      className="text--subtitle"
    >
      Â¡Conoce todos los personajes!
    </p>
  </div>
</div>
`;

exports[`Test del Home Home: Comprobando Link de ReactRouter 1`] = `
<a
  href="/characters"
  onClick={[Function]}
/>
`;
